src/Development/Rattle/Server.hs

```diff
@@ -124,6 +127,7 @@ withRattle options@RattleOptions{..} act = withUI rattleUI (return "Running") $
         timer <- offsetTime
         let s0 = S Map.empty [] emptyHazardSet [] [] speculatable Nothing
         state <- newVar $ Right $ ensureS s0
+        outFile <- newVar =<< maybe (return stdout) (`openFile` WriteMode) rattleOut
```

Comment:
I think you should be able to use a withFile block scoping over the entire rattle execution, thus avoiding the Var entirely. Perhaps `withFileOrStdout rattleOut $ \h -> `. It would also close at the end automatically and avoid the hClose's below.

Reply:
That would work in a multithreaded Rattle?